server/src/main/java/io/crate/protocols/postgres/ResultSetReceiver.java

```diff
@@ -97,9 +108,30 @@ public void allFinished() {
 
     @Override
     public void fail(@NotNull Throwable throwable) {
+        if (handleNotWritable()) return;
+
```

Comment:
Aren't these already sorta safe given that they won't queue up much data, and further processing only proceeds once the listener that gets attached to `sendErrorResponse` or `sendCommandComplete` triggers?